content/building-blocks/configure-your-computer/infrastructure-choice/configure-git.md

```diff
@@ -0,0 +1,90 @@
+---
+title: "Setting Up a New Workspace and Configuring Git "
+description: "Learn how to set up a new workspace/VM instance on Research Cloud and configure Git on the instance."
+keywords: "surf, rd, rclone, large files, webdav, store, storage, git, configure"
+weight: 3
+author: "Roshini Sudhaharan"
+authorlink: "https://nl.linkedin.com/in/roshinisudhaharan"
+draft: false
+aliases:
+  - /setup/research-drive
+---
+
+# Overview
+
+This building block takes you through the steps to configure Git on a new virtual machine (VM) on Research Cloud once you have finished [setting up your account](https://tilburgsciencehub.com/configure/research-cloud/?utm_campaign=referral-short). Follow [these steps](https://servicedesk.surf.nl/wiki/display/WIKI/Start+a+simple+workspace) to create a workspace and [log in](https://servicedesk.surf.nl/wiki/display/WIKI/Log+in+to+your+workspace) to the instance.
+
+## Setting up SSH key authentication
+
+### Step 1: Create your SSH keys
+
+You can access and write data in remote repositories on Github using SSH (Secure Shell Protocol). When you connect via SSH, you authenticate using a private key file on your local machine. To create the SSH key:
+
+- Go to the directory `~/.ssh` using `cd .ssh` command on the terminal and create a ‘config’ file using `touch config` command.
+- Then, create your SSH keys with the `ssh-keygen` command. Click enter to save the key in the default directory specified or mention an alternative directory if you would like to save it elsewhere. Then you may enter a passphrase for your private key which gives an additional layer of security for the private key.
+- Now, we have generated two keys that are required for SSH authentication: private key (id_rsa) and the public key (id_rsa.pub).
+
+
+
+  <video width="500" height="300" controls>
+    <source src="../img/ssh-keygen.mov" type="video/mp4">
+  </video>
+
+### Step 2: Configure SSH
+If you configure multiple keys for an SSH client and connect to an SSH server, the client can try the keys one at a time until the server accepts one but this process desn’t work because of how Git SSH URLs are structured. Hence, you must configure SSH to explicitly use a specific key file. To do this, edit your `~/.ssh/config` file using the `nano` command and copy-paste the following and press F3 to save.
+
+{{% codeblock %}}
+```bash
+# Host github.com
+#   User git
+#   Hostname github.com
+#   IdentityFile ~/.ssh/id_rsa
```

Comment:
have you tried adding git credentials to your persistent drive (i.e under /data/volume_2)? if you do this, then you can use git on any workspace that has access to this persistent drive, e.g if you delete your current VM and mount this drive to a new VM, you don't need to configure git from scratch. maybe you can add this as a tip below this section?

Reply:
Added this as tip! @shrabasteebanerjee